src/test/ui/regions/region-object-lifetime-in-coercion.stderr

```diff
@@ -14,40 +14,54 @@ LL | fn b(v: &[u8]) -> Box<dyn Foo + 'static> {
 LL |     Box::new(v)
    |     ^^^^^^^^^^^ lifetime `'static` required
 
-error[E0621]: explicit lifetime required in the type of `v`
-  --> $DIR/region-object-lifetime-in-coercion.rs:21:5
+error: cannot infer an appropriate lifetime
+  --> $DIR/region-object-lifetime-in-coercion.rs:20:14
    |
-LL | fn c(v: &[u8]) -> Box<dyn Foo> {
-   |         ----- help: add explicit lifetime `'static` to the type of `v`: `&'static [u8]`
-...
 LL |     Box::new(v)
-   |     ^^^^^^^^^^^ lifetime `'static` required
+   |     ---------^-
+   |     |        |
+   |     |        ...but this borrow...
+   |     this is `'static`...
+   |
+note: ...can't outlive the anonymous lifetime #1 defined on the function body at 17:1
+  --> $DIR/region-object-lifetime-in-coercion.rs:17:1
+   |
+LL | / fn c(v: &[u8]) -> Box<dyn Foo> {
+LL | |     // same as previous case due to RFC 599
+LL | |
+LL | |     Box::new(v)
+LL | | }
+   | |_^
+help: you can add a bound to the returned `dyn Trait` to make it last less than `'static` and match the anonymous lifetime #1 defined on the function body at 17:1
+   |
+LL | fn c(v: &[u8]) -> Box<dyn Foo + '_> {
+   |                               ^^^^
```

Comment:
Should I change this to be "to make it <b>require</b> less than `'static`" instead?

Reply:
I might say

help: to permit non-static references in a `dyn Trait` value, you need to add an explicit bound, as shown

or 

help: consider adding an explicit bound to the `dyn Trait` as shown, which is used to declare that the `dyn Trait` may capture non-static references

Reply:
Is the "as shown" something you'd like to add to all suggestions? In my mind it is implied always, and might not work nicely with the presentation in 3rd party tools (non-CLI uses).

I'm cooking up a commit to make this be "to permit non-static references in a `dyn Trait` value, you can add an explicit bound". WDYT?

Reply:
you can leave out the "as shown", I'm not sure how implicit it really is, but I can believe it